substrate-archive-backend/src/database.rs

```diff
@@ -75,27 +85,24 @@ impl ReadOnlyDatabase {
             }
         }
     }
+}
 
-    pub fn iter<'a>(&'a self, col: u32) -> impl Iterator<Item = KeyValuePair> + 'a {
-        self.inner.iter(col)
+impl ReadOnlyDatabaseTrait for ReadOnlyDatabase {
```

Comment:
Now that we have a `ReadOnlyDatabaseTrait`, we should rename the `ReadOnlyDatabase` to something else. Perhaps `ReadOnlyRocksDB` or `RockDBSecondary` or something (not sure if i'm 100% happy with either of those) and then rename `ReadOnlyDatabaseTrait` to `ReadOnlyDatabase`. Should make it clearer on what's actually happening